x/executionlayer/client/cli/query.go

```diff
@@ -24,92 +28,107 @@ func GetExecutionLayerQueryCmd(cdc *codec.Codec) *cobra.Command {
 		GetCmdQuery(cdc),
-		GetCmdQueryWithHash(cdc),
 	)...)
 	return executionlayerQueryCmd
 }
 
 // GetCmdQueryBalance is a getter of the balance of the address
 func GetCmdQueryBalance(cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
-		Use:   "getbalance [address]",
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("getbalance [--name \"readable_id\" or --pubkey \"secp256k1_pubkey\" or --%spub \"bech32\"] [--blockhash]", sdk.Bech32MainPrefix),
 		Short: "Get balance of address",
-		Args:  cobra.ExactArgs(1),
+		Long: fmt.Sprintf("Get balance of address.\nIt needs at least one of \"--name\", \"--pubkey\", or \"--%[1]spub\" parameter.\n"+
+			"\t--name: readabld ID\n"+
+			"\t--pubkey: Compressed Secp256k1 public key\n"+
+			"\t--%[1]spub: Bech32 encoded public key starting from '%[1]s'\n"+
+			"If you need the value of specific time, use \"--blockhash\" option.\n"+
+			"\t--blockhash: Hex blockhash value which represents of the specific time. (Default: the latest state)", sdk.Bech32MainPrefix),
+		Args: cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
-			addr, err := sdk.AccAddressFromBech32(args[0])
-			if err != nil {
-				fmt.Println("Malformed address - ", args[0])
-				fmt.Println(err)
-				return nil
-			}
 
-			name := types.ToPublicKey(addr)
-			queryData := types.QueryGetBalance{
-				Address: name,
+			// Get public key with 3 different types
+			var pubkey secp256k1.PubKeySecp256k1
+
+			if name := viper.GetString(client.FlagName); name != "" {
+				// --name: from readable id
+				pubkeyPtr, err := cliutil.GetPubKey(cdc, cliCtx, name)
+				if err != nil {
+					return err
+				}
+				pubkey = *pubkeyPtr
+			} else if rawPubkey := viper.GetString(FlagPubKey); rawPubkey != "" {
+				// --pubkey: from raw secp256k1 public key
+				pubkeyPtr, err := sdk.GetSecp256k1FromRawHexString(rawPubkey)
+				if err != nil {
+					return err
+				}
+				pubkey = *pubkeyPtr
+			} else if bech32Pubkey := viper.GetString(FlagBech32PubKey); bech32Pubkey != "" {
+				// --[bech32_prefix]pub: from bech32 public key (fridaypubxxxxxx...)
+				rawPubkey, err := sdk.GetSecp256k1FromBech32AccPubKey(bech32Pubkey)
+				if err != nil {
+					return err
+				}
+				pubkey = *rawPubkey
+			} else {
+				return fmt.Errorf("at least one of --name, --pubkey, or --%[1]spub is essential", sdk.Bech32MainPrefix)
 			}
-			bz := cdc.MustMarshalJSON(queryData)
 
-			res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/querybalance", types.ModuleName), bz)
-			if err != nil {
-				fmt.Printf("No balance data - %s \n", args[0])
-				return nil
+			var out types.QueryExecutionLayerResp
+			if blockhashstr := viper.GetString(FlagBlockHash); blockhashstr != "" {
+				blockHash, err := hex.DecodeString(blockhashstr)
+				if err != nil || len(blockHash) != 32 {
+					fmt.Println("Malformed block hash - ", blockhashstr)
+					fmt.Println(err)
+					return nil
```

Comment:
I wonder this line should be treated as an error, since users are responsible for a malformed input.

Reply:
I think that user input error is also error, isn't it?

Reply:
User input error is also err, isn't it?

Reply:
I mean that why does this function return the nil rather than the err~?